internal/providers/terraform/terragrunt_hcl_provider.go

```diff
@@ -361,6 +369,53 @@ func (p *TerragruntHCLProvider) prepWorkingDirs() ([]*terragruntWorkingDirInfo,
 	return workingDirsToEstimate, nil
 }
 
+// isParentTerragruntConfig checks if a terragrunt config entry is a parent file that is referenced by another config
+// with a find_in_parent_folders call. The find_in_parent_folders function searches up the directory tree
+// from the file and returns the absolute path to the first terragrunt.hcl. This means if it is found
+// we can treat this file as a child terragrunt.hcl.
+func (p *TerragruntHCLProvider) isParentTerragruntConfig(parent string, configFiles []string) bool {
+	for _, file := range configFiles {
+		if !isChildDirectory(parent, file) {
+			continue
+		}
+
+		file, err := os.Open(file)
```

Comment:
Do we need to close these files? Or should it not be that many to care as the execution is usually quick?

Reply:
Yes we should, for some reason the `file.Close` wasn't being recognized with my specific build tags so it got removed with `go fmt` I'm adding back in. Good spot.